### PR TITLE
Remove prop types checking in ReactCompositeComponent

### PR DESCRIPTION
**Pull request by keyanzhang**
Implements #6791. I renamed `_checkPropTypes()` to `_checkContextTypes()` since it only checks context types for now. 
Moreover, now `ReactElementValidator`'s `checkPropTypes()` and `ReactCompositeComponent`'s `_checkContextTypes()` are pretty much the same and we can merge them. Should I create a separate PR to refactor them or just change them here? @spicyj

### CODE REVIEW

File: src/renderers/shared/stack/reconciler/ReactCompositeComponent.js [hunk 1]
```diff
@@ -830,7 +794,7 @@ var ReactCompositeComponentMixin = {
       // warning for DOM component props in this upgrade
       nextProps = nextParentElement.props;
     } else {
-      nextProps = this._processProps(nextParentElement.props);
+      nextProps = nextParentElement.props;
       willReceive = true;
     }
```
Comment (spicyj) [inline src/renderers/shared/stack/reconciler/ReactCompositeComponent.js]: Can you simplify this to

```
nextProps = nextParentElement.props;
if (prevParentElement !== nextParentElement) {
  willReceive = true;
}
```

now?

File: src/renderers/shared/stack/reconciler/ReactCompositeComponent.js [hunk 2]
```diff
@@ -667,39 +667,14 @@ var ReactCompositeComponentMixin = {
...
-          newProps,
-          ReactPropTypeLocations.prop
-        );
-      }
-    }
-    return newProps;
-  },
-
-  /**
    * Assert that the props are valid
```
Comment (spicyj) [inline src/renderers/shared/stack/reconciler/ReactCompositeComponent.js]: Update this sentence?

File: src/renderers/shared/stack/reconciler/ReactCompositeComponent.js [hunk 3]
```diff
@@ -724,23 +699,12 @@ var ReactCompositeComponentMixin = {
...
-              'Failed Context Types: %s%s',
-              error.message,
-              addendum
-            );
-          }
+          warning(
+            false,
+            'Failed Context Types: %s%s',
+            error.message,
+            addendum
```
Comment (spicyj) [inline src/renderers/shared/stack/reconciler/ReactCompositeComponent.js]: Do you mind switching this to use getStackAddendumByID to show the stack? That can be a separate PR if you prefer.

**Comment by spicyj** [comment]
It's not obvious to me whether the two methods can be easily combined because they behave a little differently with the addendum (and it would be nice to calculate the addendum only if there is a warning) but if you see a way to clean them up let's do that in a follow-up PR.

**Comment by spicyj** [comment]
After fixing the inline comments, feel free to merge.

**Comment by facebook-github-bot-1** [comment]
@keyanzhang updated the pull request.
